graphql/chat/mutations.ts

```diff
@@ -79,18 +80,37 @@ export class MutateChatResolver {
 
     @Mutation(() => String)
     @Authorized(Role.PUPIL)
-    async prospectChatCreate(@Ctx() context: GraphQLContext, @Arg('instructorUserId') instructorUserId: string) {
+    async prospectChatCreate(@Ctx() context: GraphQLContext, @Arg('instructorUserId') instructorUserId: string, @Arg('subcourseId') subcourseId: number) {
         const { user: prospectUser } = context;
         const instructorUser = await getUser(instructorUserId);
 
         const conversationInfos: ConversationInfos = {
             custom: {
-                type: 'prospect',
+                ...(subcourseId && { subcourse: [subcourseId] }),
             },
         };
 
-        const conversation = await getOrCreateConversation([prospectUser, instructorUser], conversationInfos);
+        const conversation = await getOrCreateOneOnOneConversation([prospectUser, instructorUser], conversationInfos, ContactReason.PROSPECT, subcourseId);
 
         return conversation.id;
     }
+
+    @Mutation(() => String)
+    @Authorized(Role.USER)
+    async contactChatCreate(@Ctx() context: GraphQLContext, @Arg('contactUserId') contactUserId: string) {
+        const { user } = context;
+        const contactUser = await getUser(contactUserId);
+        const myContacts = await getMyContacts(user);
+        const contact = myContacts.find((c) => c.user.userID === contactUserId);
```

Comment:
What if no contact was found in the myContacts because somebody just lost the contact status or somebody is hitting the api?